src/boot.lisp

```diff
@@ -281,6 +281,25 @@
                                   (list (first v) (third v))))
                            varlist)))))))
 
+(defmacro declare (&rest declarations)
```

Comment:
This was to bypass errors in which DECLARE was being assumed as a function during bootstrap. Agreed, it should be a special form that lambda/macro processors pull out and handle.